### PR TITLE
Fix showing issues in your repositories

### LINKED ISSUE #18913: Opening an issue in another repo shows all of them as "in my repos"
### Gitea Version

1.16.2 + 1.17.0+dev-213-g5b0cdd615

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

codeberg deployment + try.gitea.io

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Description

if I

- open an issue in another repo
- it is displayed in my personal issues (`/issues` on server)
- the whole repo is shown as "In my repositories" along with all other issues of that repo
- I now see 3500+ issues more than I want to see on Codeberg without any connection to them (not subscribed, not commented, not watched or starred the repo)

As I didn't find any notice of this behaviour in the changelog, I believe this is a bug.

Reproduce: Just create an issue in a repo that already has issues and see all of them in your Issues view. I think a similar weirdness is going on with pull requests, though I don't completely understand it yet. (I only have two repos more than I want to see, I didn't open pull requests there as far as I know, but if it was showing everything where I opened an isuse, I'm also missing some projects where I opened an issue.) 

### Screenshots
grafik ← I don't know why I see those (and more!) repos.

### PR DESCRIPTION
**Pull request by Gusted**
- Make a restriction on which issues can be shown based on if you the user or team has write permission to the repository.
- Fixes a issue whereby you wouldn't see any associated issues with a specific team on a organization if you wasn't a member(fixed by zeroing the User{ID} in the options).
- Resolves #18913

@fnetX My local test instance isn't that complex as codeberg's, so I could've missed a edge case. Could you test this PR and see if this fixes the issue?

### CODE REVIEW

**Comment by Gusted** [comment]
Yeah the current wording is bad, I've pushed a comment which makes the conditions more clearly.

**Comment by fnetX** [comment]
> My local test instance isn't that complex as codeberg's, so I could've missed a edge case. Could you test this PR and see if this fixes the issue?

Yeah probably, but it seemed to be reproducibly simply. I for myself can test this at latest Monday.

File: models/issue.go [hunk 1]
```diff
@@ -1734,7 +1735,7 @@ func GetUserIssueStats(opts UserIssueStatsOptions) (*IssueStats, error) {
 	}
 
 	switch opts.FilterMode {
-	case FilterModeAll:
+	case FilterModeAll, FilterModeYourRepositories:
```
Comment (lunny) [inline models/issue.go:1793]: I think this is a break change, isn't that?

Comment (Gusted) [reply models/issue.go:1793]: I don't think so. The current behavior for the "Your repositories" was actually not limiting it to the user's repositories and rather get all issues made by the user.

**Review by fnetX — APPROVED** [review]
Tested on Codeberg.org, seems to do what it should :)

File: routers/web/user/home.go [hunk 2]
```diff
@@ -420,6 +422,36 @@ func buildIssueOverview(ctx *context.Context, unitType unit.Type) {
...
+		Private:    true,
+		AllPublic:  false,
+		AllLimited: false,
+	}
+
+	if ctxUser.IsOrganization() && ctx.Org.Team != nil {
+		repoOpts.TeamID = ctx.Org.Team.ID
+	}
+
+	userRepoIDs, _, err := models.SearchRepositoryIDs(repoOpts)
```
Comment (lunny) [inline routers/web/user/home.go]: We should not really read all the IDs into memory, we could build a subquery condition via https://gitea.com/xorm/builder to make this better performance. I have sent a PR removed that. I don't think we need to take it back.

Comment (Gusted) [reply routers/web/user/home.go]: Which PR?

Comment (lunny) [reply routers/web/user/home.go]: Can't find that one. But we really should avoid to load ids into the memory. You never know how many repositories one has.

Comment (Gusted) [reply routers/web/user/home.go]: I think I fail to understand the importance why this should be avoided. I can think of far bigger things that are loaded into memory than just a slice of int. Unless somehow we need to handle use-cases whereby a user has a good 100K of repo's(which has it's own problems) I don't think it makes sense to optimize this away, there are other things in this particular codepath that will be quite larger.

If we really should avoid this, then this PR will become quite larger than it is already as it loading this kind of information into memory and then passing it as a value is quite normal, creating cross-functions-conditions seems to be new for Gitea.

Comment (lunny) [reply routers/web/user/home.go]: In fact, you can find function `SearchRepositoryCondition` which will return a condition, you can use that condition directly. It's not that complicated as you think.

Comment (lunny) [reply routers/web/user/home.go]: We can have a `RepoCond` field in `UserIssueStatsOptions` to store the condition.

Comment (Gusted) [reply routers/web/user/home.go]: The `UserIssueStatsOptions` won't be a problem as that is one and all database queries.

The problem is rather with `issueIDsFromSearch` it gives `indexer.Search` the repo id's as a slice and looking at all the indexers, it isn't possible to get that working with database conditions.

Comment (lunny) [reply routers/web/user/home.go]: Yes, that's another problem we currently cannot handler.

File: routers/web/user/home.go [hunk 3]
```diff
@@ -430,6 +456,19 @@ func buildIssueOverview(ctx *context.Context, unitType unit.Type) {
 		opts.MentionedID = ctx.User.ID
 	case models.FilterModeReviewRequested:
 		opts.ReviewRequestedID = ctx.User.ID
+	case models.FilterModeYourRepositories:
+		if ctxUser.IsOrganization() && ctx.Org.Team != nil {
+			// Fixes a issue whereby the user's ID would be used
+			// to check if it's in the team(which possible isn't the case).
+			opts.User = nil
+		}
+		userRepoIDs, _, err := models.SearchRepositoryIDs(repoOpts)
```
Comment (lunny) [inline routers/web/user/home.go:464]: should we also use `RepoCond`?

Comment (Gusted) [reply routers/web/user/home.go:464]: That's the problem I talked about here: https://github.com/go-gitea/gitea/pull/18916#discussion_r816133809

The userRepoIDs is passed into `issueIDsFromSearch` as a option which explicitly requires the repoID's as a slice in order to pass it down into the indexer.

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
-> #19191

**Pull request merged**